http3/context_extension_type.go

```diff
@@ -0,0 +1,24 @@
+package http3
+
+import "fmt"
+
+const (
+	ContextExtensionTypeCloseCode ContextExtensionType = 0x00
+	ContextExtensionTypeDetails   ContextExtensionType = 0x01
+)
+
+// A ContextExtensionType represents an HTTP datagram context extension type code.
+// https://www.ietf.org/archive/id/draft-ietf-masque-h3-datagram-03.html#name-context-extension-types
+type ContextExtensionType uint64
```

Comment:
This type seems entirely unused.

Reply:
`CapsuleType` and `ContextExtensionType` are intended to support the [HTTP/3 MASQUE datagram draft](https://www.ietf.org/archive/id/draft-ietf-masque-h3-datagram-03.html), which the WebTransport draft is converging against.

This can be deferred to a future PR.